web/federate.go

```diff
@@ -0,0 +1,98 @@
+package web
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"bitbucket.org/ww/goautoneg"
+	"github.com/golang/protobuf/proto"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/text"
+
+	"github.com/prometheus/prometheus/promql"
+
+	clientmodel "github.com/prometheus/client_golang/model"
+	dto "github.com/prometheus/client_model/go"
+)
+
+type Federation struct {
+	QueryEngine *promql.Engine
+}
+
+func (fed *Federation) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	req.ParseForm()
+	qs := fmt.Sprintf("{%s}", strings.Join(req.Form["match[]"], ","))
```

Comment:
I forget the name of that label sometimes.

That's be good, we don't currently seem to OR between the `match[]` values. I think there's an extra bit of logic needed, as it's all currently an AND.